plugins/modules/kea_command.py

```diff
@@ -0,0 +1,216 @@
+response:
+  description: The server JSON response.
+  returned: when available
+  type: dict
+"""
+
+import json
+import os
+import socket
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def _parse_constant(s):
+    raise ValueError('Invalid JSON: "%s"' % s)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            command=dict(type='str', required=True),
+            arguments=dict(type='dict'),
+            rv_unchanged=dict(type='list', elements='int', default=[]),
+            rv_changed=dict(type='list', elements='int', default=[]),
+            socket=dict(type='path', default='/run/kea/kea4-ctrl-socket'),
+        ),
+    )
+
+    cmd = {}
+    cmd['command'] = module.params['command']
+    if module.params['arguments'] is not None:
+        cmd['arguments'] = module.params['arguments']
+    cmdstr = json.dumps(cmd, ensure_ascii=True, allow_nan=False, indent=None, separators=(',', ':'), sort_keys=True)
+    rvok = module.params['rv_unchanged']
+    rvch = module.params['rv_changed']
+    sockfn = module.params['socket']
+
+    r = {
+        'changed': False
+    }
+    rsp = b''
+
+    if not os.path.exists(sockfn):
+        r['msg'] = 'socket (%s) does not exist' % sockfn
+        module.fail_json(**r)
+
+    with socket.socket(socket.AF_UNIX, socket.SOCK_STREAM) as sock:
+        try:
+            sock.connect(sockfn)
+        except OSError as ex:
+            r['msg'] = 'could not connect to socket (%s)' % sockfn
+            r['exception'] = ex
+            module.fail_json(**r)
+        # better safe in case anything fails…
+        r['changed'] = True
+        try:
+            sock.sendall(cmdstr.encode('ASCII'))
+        except OSError as ex:
+            r['msg'] = 'could not write to socket (%s)' % sockfn
+            r['exception'] = ex
+            module.fail_json(**r)
+
+        try:
+            while True:
+                rspnew = sock.recv(8192)
+                if len(rspnew) == 0:
+                    break
+                rsp += rspnew
+        except OSError as ex:
+            r['msg'] = 'error reading from socket (%s)' % sockfn
+            r['exception'] = ex
+            module.fail_json(**r)
+
+    if len(rsp) < 15:
+        r['msg'] = 'unrealistically short response ' + repr(rsp)
+        module.fail_json(**r)
+
+    try:
+        r['response'] = json.loads(rsp, parse_constant=_parse_constant)
+    except ValueError as ex:
+        r['msg'] = 'error parsing JSON response'
+        r['exception'] = ex
+        module.fail_json(**r)
+    if not isinstance(r['response'], dict):
+        r['msg'] = 'bogus JSON response (JSONObject expected)'
+        module.fail_json(**r)
+    if 'result' not in r['response']:
+        r['msg'] = 'bogus JSON response (missing result)'
+        module.fail_json(**r)
+    res = r['response']['result']
+    if not isinstance(res, int):
+        r['msg'] = 'bogus JSON response (non-integer result)'
+        module.fail_json(**r)
```

Comment:
Given that the response is coming from a known system, are these cases something that will, realistically, ever happen? It feels over-engineered to me.

Line 201 above can replace the tests in lines 195-200 entirely. It will pop up an exception if it is not a `dict` containing a `'result'` key.

Reply:
I have zero knowledge/experience with kea, so my level of trust will be... your level of trust. Since you clearly made that choice consciously and intentionally, I am happy to stick with that. :-)

Reply:
Despite what I wrote before, on this second reading, it hits me that: if KEA stops sending parseable JSON in the socket, the entire module will stop working. If a plugin sends something that is not JSON (and somehow is embedded in the JSON response and breaks the parsing), in that case too I think the module will have much bigger problems to deal with than how controlled or not the fail is.

Not trying to convince you: the code is good as it is - even if IMHO a tad more over-engineered than what I would do. If you are happy with that, it's good for me.